cup_of_join.py

```diff
@@ -0,0 +1,14 @@
+# Arguments :
+# args - Undefined number of lists
+# sep - random char
+# return value : list
+# This functions joins lists and between them appends the given char 'sep'
+def join(*args, sep='-'):
+    if not args:
+        return None
+
+    list1 = []
+    [(list1.extend(x), list1.append(sep)) for x in args]
```

Comment:
oh no you dont :) 
https://stackoverflow.com/questions/5753597/is-it-pythonic-to-use-list-comprehensions-for-just-side-effects

Reply:
Read about zip function and nested list comprehensions

Reply:
why zip is better then extend?